src/core/main.c

```diff
@@ -355,11 +377,12 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
         } else if (streq(key, "systemd.confirm_spawn") && value) {
 
-                r = parse_boolean(value);
+                if (arg_confirm_spawn)
+                        arg_confirm_spawn = mfree(arg_confirm_spawn);
```

Comment:
mfree() is a NOP when the argument is NULL. You can call it directly hence, don't need to check for != NULL first.